persistence/src/main/java/edu/unc/lib/dl/persist/services/transfer/FSToFSTransferClient.java

```diff
@@ -69,30 +70,77 @@ public URI transferReplaceExisting(PID binPid, URI sourceFileUri) {
 
     public URI transfer(PID binPid, URI sourceFileUri, boolean allowOverwrite) {
         URI destUri = destination.getStorageUri(binPid);
+        long currentTime = System.nanoTime();
+        Path oldFilePath = createFilePath(destUri, "old", currentTime);
+        Path newFilePath = createFilePath(destUri, "new", currentTime);
+        Path destinationPath = Paths.get(destUri);
 
         try {
             // Fill in parent directories if they are not present
             Path parentPath = Paths.get(destUri).getParent();
             Files.createDirectories(parentPath);
 
+            boolean destFileExists = Files.exists(destinationPath);
+
+            if (!allowOverwrite && destFileExists) {
+                throw new BinaryAlreadyExistsException("Failed to transfer " + sourceFileUri
+                        + ", a binary already exists in " + destination.getId() + " at path " + destUri);
+            }
+
+            // Copy/move new file
             if (source.isReadOnly()) {
-                Files.copy(Paths.get(sourceFileUri), Paths.get(destUri),
+                Files.copy(Paths.get(sourceFileUri), newFilePath,
                         allowOverwrite ? COPY_ALLOW_OVERWRITE : COPY_NO_OVERWRITE);
```

Comment:
I think we only need `COPY_NO_OVERWRITE` at this point, an overwrite shouldn't occur and if it does we would want it to fail, which is what would happen in the `move` case